resolver.go

```diff
@@ -373,11 +391,13 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			}()
 			err = co.WriteMsg(query)
 			if err != nil {
+				r.concurrentQueryDec()
```

Comment:
@aboch In line 400, concurrentQueryDec() will be called for both err = nil and err != nil because once we are out of the read() the current query handling is done, either with a success or failure. But here if the write fails we continue to the next element in the for loop. So the concurrentQueryDec() has to be done only in case of failure.